### PR TITLE
Fixes Clock Cult ending with dead cultists

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
fix: Clock Cult mode will no longer end if all the cultists die. The round will not end until the Ark is destroyed or completed. 
/:cl:

Someone who understands mode code should review this, couldn't test it. 

Also #31327 probably needs to be merged before this.

This is made less necessary with my bomb PR but nevertheless there will be close rounds and having the ark fail just because some clock nerds kicked the bucket is bad gameplay and makes no sense. From what I can tell Xhuis tried to stop this from happening but did it with the wrong proc so yes it's a fix.

### CODE REVIEW

File: code/game/gamemodes/clock_cult/clock_cult.dm [hunk 1]
```diff
@@ -172,6 +172,12 @@ Credit where due:
 		return TRUE
 	return FALSE
 
+/datum/game_mode/clockwork_cult/check_finished()
+	var/obj/structure/destructible/clockwork/massive/celestial_gateway/G = locate() in GLOB.all_clockwork_objects
```
Comment (KorPhaeron) [inline code/game/gamemodes/clock_cult/clock_cult.dm]: var/obj/structure/destructible/clockwork/massive/celestial_gateway/G = GLOB.ark_of_the_clockwork_justiciar

**Review by imtakingabreakdontatme — DISMISSED** [review]

File: code/game/gamemodes/clock_cult/clock_cult.dm (hunk 1 above)
Comment (Robustin) [reply code/game/gamemodes/clock_cult/clock_cult.dm]: Why doesn't the proc below this use it?

Comment (Robustin) [reply code/game/gamemodes/clock_cult/clock_cult.dm]: I added the GLOB to the other completion proc, please add +1 GBP for this PR

**Comment by Xhuis** [comment]
This was likely a carryover from the remnants of the old clockcult code. Indifferent to this.

**Comment by kilkun** [comment]
:-1: All gamemodes should end when the main antags die. Nobody wants to spend a boring 5 minutes whacking a portal with a toolbox.

**Comment by Bawhoppen** [comment]
@kilkun You can keep playing after all the antags are dead????

**Review by Cyberboss — APPROVED** [review]

**Comment by kilkun** [comment]
@Bawhoppen thats what i dont like

**Comment by Bawhoppen** [comment]
This is a game about RP, not a game about hunting antags. Antags very much add alot under the right circumstances, but aren't the be all end all.

**Pull request merged**